File: xtask/templates/bridge/android/bridge/FfiContractVersionGuard.kt
package com.flare.im.bridge

import com.flare.im.contract.SdkContract
import com.flare.im.contract.SdkOperations

internal object FfiContractVersionGuard {
    fun validate(value: String) {
        val actual = value.trim()
        if (actual.isEmpty()) {
            throw FlareSdkException(
                code = "contract.version_unavailable",
                message = "Native binding contract version is required.",
                operation = SdkOperations.DIAGNOSTICS_FFI_CONTRACT_VERSION,
                details = mapOf(
                    "expected" to SdkContract.FFI_CONTRACT_VERSION,
                    "transport" to "ffi",
                ),
            )
        }
        if (actual != SdkContract.FFI_CONTRACT_VERSION) {
            throw FlareSdkException(
                code = "contract.version_mismatch",
                message = "Native binding contract version $actual does not match SDK ${SdkContract.FFI_CONTRACT_VERSION}.",
                operation = SdkOperations.DIAGNOSTICS_FFI_CONTRACT_VERSION,
                details = mapOf(
                    "expected" to SdkContract.FFI_CONTRACT_VERSION,
                    "actual" to actual,
                    "transport" to "ffi",
                ),
            )
        }
    }
}

File: xtask/templates/bridge/apple/Bridge/FfiContractVersionGuard.swift
import Foundation

enum FfiContractVersionGuard {
    static func validate(_ value: String) throws {
        let actual = value.trimmingCharacters(in: .whitespacesAndNewlines)
        if actual.isEmpty {
            throw FlareSdkException(
                code: "contract.version_unavailable",
                message: "Native binding contract version is required.",
                operation: SdkOperations.diagnosticsFfiContractVersion,
                details: [
                    "expected": SdkContract.ffiContractVersion,
                    "transport": "ffi",
                ]
            )
        }
        if actual != SdkContract.ffiContractVersion {
            throw FlareSdkException(
                code: "contract.version_mismatch",
                message: "Native binding contract version \(actual) does not match SDK \(SdkContract.ffiContractVersion).",
                operation: SdkOperations.diagnosticsFfiContractVersion,
                details: [
                    "expected": SdkContract.ffiContractVersion,
                    "actual": actual,
                    "transport": "ffi",
                ]
            )
        }
    }
}
